src/components/modal/Modal.svelte

```diff
@@ -0,0 +1,408 @@
+                class:confirm-body-wrapper--closeable="{closeable}">
+                <div class="ant-modal-confirm-body">
+                  {#if icon}
+                    <svelte:component this="{icon}" />
+                  {/if}
+
+                  {#if title}
+                    <span class="ant-modal-confirm-title">{title}</span>
+                  {/if}
+
+                  <div class="ant-modal-confirm-content">
+                    {#if typeof content === 'function'}
+                      <svelte:component this="{content}" />
+                    {:else}{content}{/if}
+                  </div>
+                </div>
+                <div class="ant-modal-confirm-btns">
+                  {#if modalType === 'confirm'}
+                    <Button
+                      on:click="{onCancelLocal}"
+                      {...cancelButtonProps}
+                      focusOnMount="{autoFocusButton === 'cancel'}">
+                      {cancelText}
+                    </Button>
+                  {/if}
+                  <Button
+                    type="{okType}"
+                    on:click="{onOkLocal}"
+                    {...okButtonProps}
+                    loading="{confirmLoading}"
+                    focusOnMount="{autoFocusButton === 'ok'}">
+                    {okText}
+                  </Button>
+                </div>
+              </div>
+            {/if}
+          </div>
+          {#if footer && modalType === 'standard'}
+            <div class="ant-modal-footer">
+              <slot name="footer">
+                <div>
+                  <Button
+                    on:click="{onCancelLocal}"
+                    {...cancelButtonProps}
+                    focusOnMount="{autoFocusButton === 'cancel'}">
+                    {cancelText}
+                  </Button>
+                  <Button
+                    {...okButtonProps}
+                    type="{okType}"
+                    on:click="{onOkLocal}"
+                    loading="{confirmLoading}"
+                    focusOnMount="{autoFocusButton === 'ok'}">
+                    {okText}
+                  </Button>
+                </div>
+              </slot>
+            </div>
+          {/if}
+        </div>
+      </div>
+    </div>
+  </div>
+{/if}
+
+<script>
+  import { createEventDispatcher, onMount, onDestroy } from "svelte";
+  import { fade, fly } from "svelte/transition";
+  import {
+    CloseOutlined,
+    InfoCircleOutlined,
+    CheckCircleOutlined,
+    CloseCircleOutlined,
+    ExclamationCircleOutlined
+  } from "@/components/icons";
+  import Button from "../button/Button.svelte";
+  import { modalDestroyFunctions } from "./store";
+
+  const dispatch = createEventDispatcher();
+
+  // ********************** Props **********************
+
+  // If the modal is visible
+  export let visible = false;
+  // User has the option to remove the footer
+  export let footer = true;
+  // Modal width
+  export let width;
```

Comment:
Do we want a default value for `width`, if we don't specify a default value for props `svelte` will consider them required props.

Reply:
So it gets set here https://github.com/svelting/svant/pull/15/files#diff-5d2163e0141e01baa5081921b3ce9d9eR291

However it's a good habit to be in. I'll set it to null initially.